src/app/components/signup/signup.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, inject } from '@angular/core';
+import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { AuthService } from '../../services/auth.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-signup',
+  standalone: true,
+  imports: [ReactiveFormsModule],
+  templateUrl: './signup.component.html',
+  styleUrl: './signup.component.css',
+})
+export class SignupComponent {
+  fb = inject(FormBuilder);
+  authService = inject(AuthService);
+  router = inject(Router);
+
+  form = this.fb.nonNullable.group({
+    username: ['', Validators.required],
+    email: ['', Validators.required],
```

Comment:
there is also a built in validator called `validators.email` that checks emails for you

Reply:
All my form validation and styling will be done on my feature-forms branch. So its in progress. Thank you! I will remember this when I do it.